### ISSUE TITLE
Class 'barrelstrength\sproutbasereports\base\DataSource' not found

### ISSUE THREAD

**Issue by johnj24** [barrelstrength/craft-sprout-email#144]
### Description (screenshots, logs, ...)
I've just upgraded a client site from Craft 2.6.3015 to 3.4.9 and now when I try to access one of my email notifications, I get the following error message:

`Class 'barrelstrength\sproutbasereports\base\DataSource' not found`

### Steps to reproduce

1. Upgrade from Craft 2.6.3015 to 3.4.9
2. Go to Sprout Email > Notifications > Click on a notification

### Additional info

- Craft version: 3.4.9
- PHP version: 7.1.0
- Database driver & version:  MySQL 5.7.24
- Plugins & versions:

_For some issues, we may request that you send the following items to Support <sprout@barrelstrengthdesign.com>. If you feel your issue may fall in that category, feel free to send things over after you post this issue and reference the issue in your email._

- `composer.json`
- `storage/logs/web.log` (or a zip of the whole log folder)
- An export of your database (or, if you can't share the whole db, the specific Sprout tables)

**Comment by BenParizek** [comment]
@johnj24 It appears the most recent version of Craft 2.x is Craft 2.9.2. Can you confirm that you upgraded to Craft 2.9.2 and all the latest version of all Craft 2.x available plugins before you upgraded to Craft 3?

**Comment by johnj24** [comment]
@BenParizek I updated to Craft 2.9.2 and Sprout Email 3.1.3 before upgrading, but unfortunately I'm getting the same error.

**Comment by rickbergmans** [comment]
I've got the exact same problem. I updated from Craft 3.1.x to Craft 3.4.9. None of my notifications work and I can't access any of them to debug.

**Comment by rickbergmans** [comment]
Even after completely uninstalling, I still get this error. All notifications for a community website are broken now. Is there something I could try myself to (temporarily) resolve this?

<img width="1498" alt="afbeelding" src="https://user-images.githubusercontent.com/17927472/77047172-9c006700-69c4-11ea-99cc-3c1f5c69a0d4.png">

**Comment by rickbergmans** [comment]
Notifications weren't sending because of Twig trying to render my CSS styles in my email template. I've fixed that, but editing notifications in Craft still doesn't work.

**Comment by BenParizek** [comment]
@johnj24 @rickbergmans I haven't been able to recreate this yet but let me describe a bit about what's happening on this page and maybe we can get to the bottom of it.

When the Notifications page loads the code checks a value in the `barrelstrength\sproutbasereports\base\DataSource` class and it appears in your installs that class is unavailable and causing this error.

I don't know why that class would be unavailable but let's dig in and see what we learn:

The missing class is located in the folder`vendor/barrelstrength/sprout-base-reports/src/base/DataSource`. Can you check and see if you see that file there?

It also may be worth just re-running composer install to make sure you have all the latest packages. If something did get tripped up in the upgrade, you can do the following to make sure you have all the most recent packages:

1. Review your `composer.json` file and make sure you have the latest version of Craft each plugin listed in the `require` array
2. Delete the `composer.lock` file
3. Delete the `vendor` folder
4. Re-run `composer update`

Let me know if that helps get things resolved or if you still an error.

**Comment by johnj24** [comment]
@BenParizek I'm not even seeing a `sprout-base-reports` folder inside of `vendor/barrelstrength`

Does Sprout Reports need installed for that folder to be available?

**Comment by BenParizek** [comment]
@johnj24 Interesting. `sprout-base-reports` should be [included with Sprout Forms as we require it in the composer.json file](https://github.com/barrelstrength/craft-sprout-forms/blob/v3.8.7/composer.json#L32) (it does not require Sprout Reports to be installed, it's a shared library used both with Sprout Forms and Sprout Reports, whichever you are using). You may want to start by re-running the composer install steps listed in my last message.

I'm not even sure what you would have to do to not have Sprout Base Reports installed. Would you be able to share what steps you took to get Sprout Forms installed in your project? Composer shouldn't be able to install the `sprout-forms` package without errors if the `sprout-base-reports` package was not available.

**Comment by johnj24** [comment]
@BenParizek I don't think I did anything out of the ordinary on my upgrade. I followed the Craft 3 upgrade guidelines. However, its happened twice so maybe somethings up? I could send you the backup database or whatever you need.

Meanwhile, let me try the composer re-run you outlined and see if that helps.

**Comment by johnj24** [comment]
@BenParizek Just re-ran `composer update` after deleting `/vendor/` and `composer.lock`. It's not installing the `sprout-base-reports` folder. Here's the Barrel Strength specific CLI readout:

```
  - Installing barrelstrength/sprout-base-email (1.2.6): Downloading (100%)
  - Installing barrelstrength/sprout-base (5.1.2): Downloading (100%)
  - Installing barrelstrength/sprout-email (4.1.5): Downloading (100%)
  - Installing barrelstrength/sprout-base-import (1.0.6): Downloading (100%)
  - Installing barrelstrength/sprout-fields (3.6.1): Downloading (100%)
```

**Comment by BenParizek** [comment]
@johnj24 Can you share your `composer.json` file?

**Comment by johnj24** [comment]
```
{
  "require": {
    "aelvan/imager": "v2.4.0",
    "barrelstrength/sprout-email": "4.1.5",
    "barrelstrength/sprout-fields": "3.6.1",
    "craftcms/cms": "^3.4.0",
    "craftcms/redactor": "2.6.0.1",
    "ether/simplemap": "3.8.2",
    "nystudio107/craft-seomatic": "3.2.47",
    "topshelfcraft/wordsmith": "^3.1",
    "vlucas/phpdotenv": "^3.4.0"
  },
  "require-dev": {
    "yiisoft/yii2-shell": "^2.0.3"
  },
  "autoload": {
    "psr-4": {
      "modules\\": "modules/"
    }
  },
  "config": {
    "sort-packages": true,
    "optimize-autoloader": true,
    "platform": {
      "php": "7.0"
    }
  },
  "scripts": {
    "post-root-package-install": [
      "@php -r \"file_exists('.env') || copy('.env.example', '.env');\""
    ]
  }
}
```

**Comment by BenParizek** [comment]
@johnj24 I'm sorry. I was looking right past the issue here thinking I was troubleshooting Sprout Forms not Sprout Email. It appears I overlooked adding the `sprout-base-reports` dependency to Sprout Email when we released that new feature recently.

@johnj24 @rickbergmans I've just released Sprout Email v4.1.6 which adds the `sprout-base-reports` dependency and should resolve this issue.
